charter/index.html

```diff
@@ -191,7 +191,7 @@ <h2>Motivation and Background</h2>
         <h2>Scope</h2>
 
         <p>
-          <span class=new>The scope of the Working Group is to define a web protocol for use between client-side applications, on the one hand, and storage and identity servers, on the other, allowing</span> users to keep authority over their data, identity, and privacy.
+          <span class=new>The scope of the Working Group is to define a web protocol that client-side applications can use to authenticate users and store their data with user-chosen identity and storage servers. Applications that use this protocol can more easily allow their</span> users to keep authority over their data, identity, and privacy.
```

Comment:
This looks good to me. I'm also fine with @melvincarvalho's tweak to broaden it to applications in general.

Reply:
I quite like @TallTed's version. If we want to keep something closer to the original sentence, I would suggest at least *s/allow/respect/* since it is the user deciding where their data gets stored, not a decision of any of the applications the user currently chooses to access the user's data with.

Reply:
The scope for defining such a web protocol doesn't need to tie itself to identity providers for authenticating users, which typically rely on third parties. I suggest not constraining the scope in this way to allow different approaches and to make the protocol specification more adaptable and flexible.

+1 to @jyasskin 's suggestion, -1 to @TallTed 's suggestion.

Reply:
> [@csarven] typically rely on third parties

That's a historical perspective, and need not play a part in the design being contemplated here. _User-chosen identity providers_ are (or should be, or at least could be) rather different beasts than _service-provider-imposed identity providers_.

I might suggest you take a look at the current work of the WICG and the FedID CG.

Reply:
My emphasis was on the broader notion of "identity providers" rather than focusing solely on server-imposed ones.

My point still stands: the initially proposed text in this PR aligns with the intended meaning of the prior text and understanding. Your suggestion introduces a new constraint that diverges from that.

Reply:
`authenticate users and store their data with user-chosen identity and storage servers` includes the notion of "identity providers" though it calls them "identity servers".

Reply:
```suggestion
          <span class=new>The scope of the Working Group is to define a web protocol that applications (sometimes known as relying parties, or `RP`s) can use to authenticate users (via user-chosen identity providers, known as `IdP`s) and store their data (via user-chosen storage providers). Applications that use this protocol can more easily allow their users to exert and maintain authority over their data, identity, and privacy, by treating all of these as loosely coupled to each other, to the users, and to the applications. Such applications will be able to present and interleave data stored by multiple users in multiple data storage spaces.</span>
```

Reply:
"authenticate users and store their data with user-chosen identity and storage servers" implies a separation between the concepts of identity and storage servers. I don't see "identity provider" or "identity servers" which seems to put emphasis/place a constraint on the authentication system.

Reply:
We need to decide how many details to include. If we're going to communicate something, I would prefer to do it explicitly rather than implicitly. 
While the user wants to autheNticate with the app, the app needs authoriZation to access protected resources on the Resource Server (storage). The more complex part comes with the general case where the End-user != Resource Owner, which I see as a longer delegation chain.
There is also an aspect of the application (client) autheNticating itself with the Authorization Server.
We should convey that all those aspects need to be loosely coupled but allowed to be combined. For example, combined IdP and AS can provide a smoother user experience. Especially keeping in mind conversations like the one in https://github.com/fedidcg/FedCM/issues/477

Reply:
@csarven -- Your text is most obvious read as two phrases, 
"(authenticate users and store their data) (with user-chosen identity and storage servers)" Rather than "[implying] a separation between the concepts of identity and storage servers", it makes it seem that _user authentication and data storage_ is one thought, as is "identity and storage server(s)".

My suggestion is meant to make plain that authenticating users with user-chosen identity servers, and storing user data on user-chosen storage servers, are separate, loosely-coupled actions which *can* be provided by/through a single entity (as is typical today, though not necessarily by user choice, as it is typically the only option) which which also *can* be provided by distinct entities at each layer/point-of-service (as is *not* typical today).

The last part is the key innovation of Solid — the loosely-coupled architecture that allows a user to authenticate via their choice of identity provider, store their data on their choice of storage server, and run their choice of apps, whether on their client-device or a chosen app server, blending data that is stored on *their* server with data that is stored on the servers chosen by other users in their network (whether social, business, family, or otherwise).

Reply:
What definition of "identity provider" are you using towards authentication? ( https://en.wikipedia.org/wiki/Identity_provider or referenced therein? https://fedidcg.github.io/FedCM/#idp ? ... )  and which classes of authentication systems does it include and exclude?

There is no point in throwing "identity provider" into the charter (scope) if folks only really want particular specifications or category of authentication systems. Just say exactly what you want in that case instead of beating around the bush.

My point still stands that the suggestion to use "identity provider" is only constraining where that constraint wasn't intended to begin with.

Let me also point out that the Solid CG did in fact review authentication systems in the context of scope:

https://github.com/solid/specification/blob/main/meetings/2023-05-31.md#scope-needs-to-be-tightly-defined-with-narrow-focus

and the agreement was to omit explicit mention of mechanisms. The slip with IdP, RP etc. seems counter to me.

Here is also a snippet (from the minutes linked above) that is forward looking:

>* TBL: It has moved steadily. Stability of the Solid OIDC is important. If we think of moving from Solid OIDC like HTTPSignatures where you have public keys.. like we had with WebID-TLS would be a lot simpler and lot more powerful and secure. So I think if we move, it is worth considering moving to two where we experiment with completely different public-key based rather than HTTP redirect based.

The agreement to not constrain the scope was alone made in the context of Solid WG charter. That alone should suffice for the PUMPKIN WG and that the text in this PR should at least reflect that view or thereabouts. If the original intention isn't conveyed in this PR's text or your suggestion, then we can revise.

Reply:
I'm not trying to say that any (existing or theorized, individual or category of) identity or authentication system is a predetermined winner nor suitable for all use cases; hence the (in my mind) generic "[Identity Provider (IdP)](https://dbpedia.org/resource/Identity_provider)".

If changing that to "Identity Servers (IdS)" (which is not yet described on Wikipedia/DBpedia) will pass your review, where "Identity Providers (IdP)" seems not to, I'm OK with that change.

My language is meant to make plain that this is not about a *bundle* of identity and storage services, probably from one provider, of which a user currently must choose all-or-none, and from which they *might* be able to export their data but such exports may not be in a user-usable/service-migratable form; but rather about a *pool* of identity services and a *pool* of storage services, each pool comprised of offerings from *multiple* providers, of which a user can mix-and-match (and change from/to) as they like.

Reply:
Okay. What I'm saying is that the use of "identity provider" isn't as generic as implied. Introducing terminology that limits the intended meaning takes us in a different direction.

Can we focus on refining the original intention given the insight, context, and reference rather than interpreting or making better sense of potentially faulty text?

What passes "my review" is what was intended / consensus.

---

The issue isn't about better expressing "Identity Providers" with something like "Identity Servers". I prefer not to rehash what "identi*" implies here, but as mentioned, "provider," while not always, generally suggests a third party - an entity other than the agent with which the identity is associated. This relates to who ultimately holds the highest technical and often social authority over the "identi*" (at least according to [AWWW](https://www.w3.org/TR/webarch/)). This cuts through some of the core concerns about the user's actual agency at the end of the day.

Authentication doesn't boil down to using generic "identity providers". Example: We didn't need a "provider" with WebID-TLS. It ultimately relied on the user showing their cert on the client-side without any hops or checks through an "identity provider". Yes, a resource server for WebID Profile Documents is used, but that's beside the point (since other systems also use a server to present a description as well.)

And again, I'm not referring to presenting a variety of identity providers or services for the user to choose from at sign-in. (I don't care about that, and as I understand it, the consensus is to move away from that any way.)

---

That said, authentication is not always identity-centric either.

---

I think I've made myself clear on this topic but let me know if I can clear anything further. Handing this particular thread off to @pchampin and others... until there is new information that needs to be evaluated.

Reply:
I reported a few of the changes suggested by @TallTed in 59c37f3.
I sympathize with @csarven's concern about "identity provider" and "relying parties". I also have issues with "loosely coupled to each other, **to the users**, and to the applications". I'm not sure what "to the users" means in this context...